### ISSUE TITLE
Running hubot with grunt-shell throws readline_history error

### ISSUE THREAD

**Issue by robert-rally** [github/hubot#845]
I'm using Grunt to launch hubot using a grunt-shell command. This used to work fine, but now I get the following error:

[Tue Jan 06 2015 11:00:41 GMT-0700 (MST)] ERROR TypeError: Cannot read property 'push' of undefined
  at /hubot-test/node_modules/hubot/node_modules/readline-history/readline-history.js:112:19
  at ReadStream.onend (/hubot-test/node_modules/hubot/node_modules/readline-history/readline-history.js:52:21)
  at ReadStream.emit (events.js:117:20)
  at _stream_readable.js:943:16
  at process._tickCallback (node.js:419:13)

**Comment by technicalpickles** [comment]
@robert-rally I'm really familiar with grunt-shell. Would you be able to share a code-snippet with how you are launching it?

**Comment by robert-rally** [comment]
## Gruntfile.coffee
```coffeescript
module.exports = (grunt) ->
  grunt.loadNpmTasks 'grunt-shell'
  grunt.registerTask 'jarvis', ['shell:jarvis']

  grunt.initConfig
    shell:
      jarvis:
        command: './bin/hubot --name jarvis'
```
## bin/hubot
```shell
#!/bin/sh
npm install
export PATH="node_modules/.bin:node_modules/hubot/node_modules/.bin:$PATH"
exec node_modules/.bin/hubot "$@"
```
## package.json
```javascript
{
  "dependencies": {
    "hubot": ">= 2.6.0 < 3.0.0",
    "hubot-scripts": ">= 2.5.0 < 3.0.0"
  },
  "devDependencies": {
    "grunt-shell": "~1.1.1"
  },
  "engines": {
    "node": ">= 0.10.x",
    "npm": ">= 2.0.x"
  }
}
```

**Comment by viking** [comment]
This also happens if you run hubot with `node debug`.

**Comment by benbalter** [comment]
I can recreate this as well. Hubot runs fine with when executed with `./bin/hubot` but errs out when executed with a Procfile of `web: bin/hubot` and then `foreman start`.

**Comment by robert-rally** [comment]
The bug is in readline_history. It's a bad monkey-patch that assumes readLine.history is always initialized. Spoiler alert: it isn't.

This history feature provides dubious value and definitely breaks things. Is it possible to simply remove it from hubot?

**Comment by technicalpickles** [comment]
> This history feature provides dubious value

It was added based on feedback actually. I find it pretty valuable in development personally, but I can't speak for anyone else

>  definitely breaks things. Is it possible to simply remove it from hubot?

Yes, it can be removed, but I would like to add something in it's place if possible

**Comment by technicalpickles** [comment]
This should be fixed in 2.11.3.
